lib/eco/grammars/grammars.py

```diff
@@ -103,6 +104,17 @@ def add_alternative(self, nonterminal, language):
     def change_start(self, name):
         self.extract = name
 
+    def set_auto_limit(self, lang, tokentype):
+        if self.auto_limit_start is None:
+            self.auto_limit_start = {}
+        self.auto_limit_start[lang] = tokentype
+
+    def auto_allows(self, lang, tokentype):
```

Comment:
This function returns `True` or `None` which seems weird? Could we simplify it to:

```python
def auto_allows(self, lang, tokentype):
  if self.auto_limit_start and self.auto_limit_start.has_key(lang):
    return tokentype in self.auto_limit_start[lang]
  return False
```

or similar?